File: csharp/DesiArchitect.DesignPatterns/07_Facade/After/PaymentService.cs
using DesiArchitect.DesignPatterns.Shared;

namespace DesiArchitect.DesignPatterns.Facade.After;

public class PaymentService : IPaymentService
{
    public void Charge(BookingRequest request)
        => Console.WriteLine($"  PaymentService: Charged ₹{request.Fare} for {request.PassengerName}");
}

File: typescript/src/07-facade/after/payment-service.ts
import type { BookingRequest } from '../../shared/models.js';
import type { IPaymentService } from './i-payment-service.js';

export class PaymentService implements IPaymentService {
  charge(request: BookingRequest): void {
    console.log(`  PaymentService: Charged ₹${request.fare} for ${request.passengerName}`);
  }
}